atspi-connection/src/p2p.rs

```diff
@@ -0,0 +1,746 @@
+//! Extends the `AccessibilityConnection` with P2P capabilities.
+
+use async_executor::Executor;
+use async_lock::Mutex;
+use atspi_common::{AtspiError, ObjectRef};
+use atspi_proxies::{
+	accessible::{AccessibleProxy, ObjectRefExt},
+	application::{self, ApplicationProxy},
+	proxy_ext::ProxyExt,
+};
+use futures_lite::{future::block_on, stream::StreamExt};
+use std::sync::Arc;
+use zbus::{
+	conn::Builder,
+	fdo::DBusProxy,
+	names::{BusName, OwnedBusName, OwnedUniqueName, OwnedWellKnownName},
+	proxy::CacheProperties,
+	zvariant::ObjectPath,
+	Address,
+};
+
+#[cfg(feature = "tracing")]
+use tracing::{debug, info, warn};
+
+use crate::AtspiResult;
+
+const ACCESSIBLE_ROOT_OBJECT_PATH: &str = "/org/a11y/atspi/accessible/root";
+const REGISTRY_WELL_KNOWN_NAME: &str = "org.a11y.atspi.Registry";
+
+/// Represents a peer with the name, path and connection for the P2P peer.
+#[derive(Clone, Debug)]
+pub struct Peer {
+	unique_name: OwnedUniqueName,
```

Comment:
You can use `zbus::BusName` here. No need for the two fields.

Reply:
No. This is not an 'or'-situation. The `Peer` will always have a `UniqueName` which may (or may not) own a `WellKnownName`. For a `WellKnownName' to make any snse it must have an owner - a `UniqueName`. 
In fact a `UniqueName` (a connection) can own as many `WellKnownName`'s as it wants. (We would represent that as having multiple `Peer`s). `WellKnownName`'s however cannot be on the bus on their own, they always need an owner - a process with a connection to the bus to communicate with.